packages/flutter/test/gestures/drag_test.dart

```diff
@@ -648,6 +649,342 @@ void main() {
+    tester.route(pointer6.move(const Offset(0.0, 100.0)));
+    log.add('-d');
+    // #6 pointer move to up 50.0, received delta should be (-50.0, 0.0).
+    tester.route(pointer6.move(const Offset(0.0, 50.0)));
+
+    // _moveDeltaBeforeFrame = { 5: Offset(50, 0), 6: Offset(50, 0),}
+
+    log.add('-e');
+    // #6 pointer move to down 1.0, received delta should be (0.0, 1.0).
+    tester.route(pointer6.move(const Offset(0.0, 51.0)));
+    log.add('-f');
+    // #6 pointer move to up 1.0, received delta should be (0.0, -1.0).
+    tester.route(pointer6.move(const Offset(0.0, 50.0)));
+
+    // _moveDeltaBeforeFrame = { 5: Offset(50, 0), 6: Offset(50, 0),}
+
+    log.add('-g');
+    // #5 pointer move to down 1.0, received delta should be (1.0, 0.0).
+    tester.route(pointer5.move(const Offset(0.0, 51.0)));
+    log.add('-h');
+    // #5 pointer move to up 1.0, received delta should be (-1.0, 0.0).
+    tester.route(pointer5.move(const Offset(0.0, 50.0)));
+
+    // _moveDeltaBeforeFrame = { 5: Offset(50, 0), 6: Offset(50, 0),}
+
+    log.add('-i');
+    // #6 pointer move to down 10.0, received delta should be (0.0, 10.0).
+    tester.route(pointer6.move(const Offset(0.0, 60.0)));
+    log.add('-j');
+    // #6 pointer move to up 10.0, received delta should be (0.0, -10.0).
+    tester.route(pointer6.move(const Offset(0.0, 50.0)));
+
+    // _moveDeltaBeforeFrame = { 5: Offset(50, 0), 6: Offset(50, 0),}
+
+    // Trigger a new frame.
+    SchedulerBinding.instance.handleBeginFrame(const Duration(milliseconds: 100));
+    SchedulerBinding.instance.handleDrawFrame();
+
+    log.add('-k');
+    // #5 pointer move to down 1.0, received delta should be (0.0, 1.0).
+    tester.route(pointer5.move(const Offset(0.0, 51.0)));
+    log.add('-l');
+    // #5 pointer move to up 1.0, received delta should be (0.0, -1.0).
+    tester.route(pointer5.move(const Offset(0.0, 50.0)));
+
+    tester.route(pointer5.up());
+    tester.route(pointer6.up());
+
+    // Tear down 'currentSystemFrameTimeStamp'
+    SchedulerBinding.instance.handleBeginFrame(Duration.zero);
+    SchedulerBinding.instance.handleDrawFrame();
+
+    expect(log, <String>[
+      '-a',
+      'drag-update (Offset(0.0, 100.0))',
+      '-b',
+      'drag-update (Offset(0.0, -50.0))',
+      '-c',
+      'drag-update (Offset(0.0, 50.0))',
+      '-d',
+      'drag-update (Offset(0.0, -50.0))',
+      '-e',
+      'drag-update (Offset(0.0, 1.0))',
+      '-f',
+      'drag-update (Offset(0.0, -1.0))',
+      '-g',
+      'drag-update (Offset(0.0, 1.0))',
+      '-h',
+      'drag-update (Offset(0.0, -1.0))',
+      '-i',
+      'drag-update (Offset(0.0, 10.0))',
+      '-j',
+      'drag-update (Offset(0.0, -10.0))',
+      '-k',
+      'drag-update (Offset(0.0, 1.0))',
+      '-l',
+      'drag-update (Offset(0.0, -1.0))',
+    ]);
+  });
+
+  testGesture('Pan drag with multiple pointers - maxAllPointers', (GestureTester tester) {
```

Comment:
I think this test case tested too much for the case that one pointer moves back and forth within a frame (we don't have to remove them, up to you), but doesn't test the case with more than 2 pointers. We want to test that the pointers that move not as much do not affect the resulting delta, and that they might become the border pointers when they move a lot.